### PR TITLE
Fix restore snapshot call

### PR DESCRIPTION
**Pull request by ismirlia**
The restore snapshot call is incorrectly labeled as "create" even though that is not the endpoint it calls, and there is a separate create snapshot call in the ibm-pi-instance.go file.

### CODE REVIEW

**Review by michaelkad — DISMISSED** [review]

**Review by michaelkad — DISMISSED** [review]

File: clients/instance/ibm-pi-snapshot.go [hunk 1]
```diff
@@ -81,8 +81,8 @@ func (f *IBMPISnapshotClient) GetAll() (*models.Snapshots, error) {
 	return resp.Payload, nil
 }
 
-// Create or Restore a Snapshot
-func (f *IBMPISnapshotClient) Create(instanceID, snapshotID, restoreFailAction string) (*models.Snapshot, error) {
+// Restore a Snapshot
+func (f *IBMPISnapshotClient) Restore(instanceID, snapshotID, restoreFailAction string) (*models.Snapshot, error) {
```
Comment (yussufsh) [inline clients/instance/ibm-pi-snapshot.go]: This is a breaking change. You can create another function Restore and call Create(). Also, you could mark it as deprecated.

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: clients/instance/ibm-pi-snapshot.go (hunk 1 above)
Comment (ismirlia) [reply clients/instance/ibm-pi-snapshot.go]: @yussufsh How do I mark elements in the power-go-client as deprecated?

File: clients/instance/ibm-pi-instance.go [hunk 2]
```diff
@@ -253,18 +253,18 @@ func (f *IBMPIInstanceClient) GetSnapShotVM(id string) (*models.Snapshots, error
 }
 
 // Restore a Snapshot of an Instance
-func (f *IBMPIInstanceClient) RestoreSnapShotVM(id, snapshotid, restoreAction string, body *models.SnapshotRestore) (*models.Snapshot, error) {
+func (f *IBMPIInstanceClient) RestoreSnapShotVM(instanceID, snapshotID, restoreFailAction string, body *models.SnapshotRestore) (*models.Snapshot, error) {
```
Comment (yussufsh) [inline clients/instance/ibm-pi-instance.go:256]: IMO appending `instance` to the ID seems redundant. The functions in InstanceClient as particularly related to Instances.
Eg: In Volume client,`id` should mean volumeID and for instance, instanceID (operations like Attach/Detach).

At many places, it is named as pvmInstanceID or pvmID or something else.

File: clients/instance/ibm-pi-snapshot.go [hunk 3]
```diff
@@ -81,12 +81,29 @@ func (f *IBMPISnapshotClient) GetAll() (*models.Snapshots, error) {
...
 
-// Create or Restore a Snapshot
-func (f *IBMPISnapshotClient) Create(instanceID, snapshotID, restoreFailAction string) (*models.Snapshot, error) {
+// Restore a Snapshot
+func (f *IBMPISnapshotClient) Restore(instanceID, snapshotID, restoreFailAction string, body *models.SnapshotRestore) (*models.Snapshot, error) {
 	params := p_cloud_p_vm_instances.NewPcloudPvminstancesSnapshotsRestorePostParams().
 		WithContext(f.ctx).WithTimeout(helpers.PICreateTimeOut).
 		WithCloudInstanceID(f.cloudInstanceID).WithPvmInstanceID(instanceID).
-		WithSnapshotID(snapshotID).WithRestoreFailAction(&restoreFailAction)
+		WithSnapshotID(snapshotID).WithRestoreFailAction(&restoreFailAction).
```
Comment (yussufsh) [inline clients/instance/ibm-pi-snapshot.go:89]: I did not mean to duplicate the code.
Just call Create() from here or vice versa.

Comment (ismirlia) [reply clients/instance/ibm-pi-snapshot.go:89]: The existing create which does a restore does not call the restore with all the proper parameters. You currently cannot pass a fail action to it.

Comment (yussufsh) [reply clients/instance/ibm-pi-snapshot.go:89]: I see, if `restoreFailAction` is required then we can skip the call.
If Create() itself is failing then we need to fix it.

**Comment by ismirlia** [comment]
Cancelling this PR.

**Pull request closed without merging**